### PR TITLE
Megh/intake

### PR DESCRIPTION
**Pull request by Meghd2cx**
Added intake functionality
Intake not tested yet
Half-implemented PID for intake system (not applied for intake, but for arm)

### CODE REVIEW

File: code/src/main/java/frc/robot/subsystems/Intake.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        // m_armPIDController.setS
+
+        // Motor is inverted
+    }
+
+    public void raiseArm() {
+        int count = armEncoder.get();
+
+        // Based on the gear ratio of the motor
+        int PulsesPerRevolution;
```
Comment (dotnetprofessional) [inline code/src/main/java/frc/robot/subsystems/Intake.java]: use camelCase for you names

File: code/src/main/java/frc/robot/subsystems/Intake.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    public void lowerArm() {
+
+    }
+
+    public boolean isLowered() {
+        boolean isArmStopped = armEncoder.getStopped();
+
+    }
+
+    public void spinIntake(boolean direction) {
```
Comment (dotnetprofessional) [inline code/src/main/java/frc/robot/subsystems/Intake.java:70]: Why was this changed from velocity?

File: code/src/main/java/frc/robot/subsystems/Intake.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        intakeController.set(scaledvelocity);
+
+    }
+
+    public void stopIntake() {
+        intakeController.set(0);
+    }
+
+    public boolean isSpinning() {
+        if (Math.abs(intakeController.get()) > 0) {
```
Comment (dotnetprofessional) [inline code/src/main/java/frc/robot/subsystems/Intake.java]: FYI you can write this as simply:

return Math.abs(intakeController.get()) > 0;

File: code/src/main/java/frc/robot/subsystems/Intake.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+import edu.wpi.first.wpilibj.PWMVictorSPX;
+import edu.wpi.first.wpilibj.SpeedController;
+import edu.wpi.first.wpilibj.controller.PIDController;
+import frc.robot.common.IIntakeSubsystem;
+import frc.robot.common.ILogger;
+import frc.robot.common.TraceableSubsystem;
+
+public class Intake extends TraceableSubsystem implements IIntakeSubsystem {
+
+    public SpeedController intakeController = new PWMVictorSPX(1);
```
Comment (dotnetprofessional) [inline code/src/main/java/frc/robot/subsystems/Intake.java]: I'm seeing a lot of magic numbers. We really need to get in the habit of adding these to the config class.

**Review by dotnetprofessional — CHANGES_REQUESTED** [review]

File: code/src/main/java/frc/robot/subsystems/Intake.java (hunk 2 above)
Comment (Meghd2cx) [reply code/src/main/java/frc/robot/subsystems/Intake.java:70]: It was changed from velocity because we simply wanted the intake to spin either way at a pre-specific velocity or to be turned off.
